### PR TITLE
add --append= switch to systemd-measure

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by bluca** [comment]
Please add tests for this - unit tests for the json helper, and integration tests in TEST-70 for the measure --append

File: src/boot/measure.c [hunk 1]
```diff
@@ -71,6 +73,7 @@ static int help(int argc, char *argv[], void *userdata) {
                "     --private-key=KEY   Private key (PEM) to sign with\n"
                "     --public-key=KEY    Public key (PEM) to validate against\n"
                "     --json=MODE         Output as JSON\n"
+               "     --append=PATH       Load specified JSON signature, and append new signature to it\n"
                "  -j                     Same as --json=pretty on tty, --json=short otherwise\n"
```
Comment (keszybz) [inline src/boot/measure.c]: `--json` and `-j` should remain adjacent.

File: man/systemd-measure.xml [hunk 2]
```diff
@@ -182,6 +182,19 @@
...
+      <varlistentry>
+        <term><option>--append=</option><replaceable>PATH</replaceable></term>
+
+        <listitem><para>When generating a PCR JSON signature (via the <command>sign</command> command),
+        combine it with a previously generated PCR JSON signature, and output it as one. The specified path
+        must refer to a regular file that contains a valid JSON PCR signature object. The specified file is
+        not modified. It will be read first, then the newly generated signature appended to it, and the
+        resulting object is written to standard output. Use this to generate a single JSON object consisting
+        from signatures made with a number of signing keys (for example, to have one key per boot phase and
+        similar). The command will suppress duplicates: if a specific signature is already included in a JSON
```
Comment (keszybz) [inline man/systemd-measure.xml]: The sentence sounds better without "and similar".

File: man/systemd-measure.xml [hunk 3]
```diff
@@ -182,6 +182,19 @@
...
+        signature object it is not added a second time.</para></listitem>
+      </varlistentry>
+
```
Comment (keszybz) [inline man/systemd-measure.xml:197]: I'd be nice to have an example in the examples section for this too.

**Review by keszybz — COMMENTED** [review]
Looks like this should work.

Maybe combine the last three commits into one? They are one logical change.

I think we should insert some text in the JSON that describes what each signature is for… Without that, it's hard to make sense of the file. But that's subject for another pull request.

**Comment by keszybz** [comment]
I'm adding support for this in #25180. Seems to work fine.

**Review by bluca — APPROVED** [review]

**Pull request merged**